src/persist-client/src/lib.rs

```diff
@@ -304,13 +308,54 @@ impl PersistClient {
             reader_id,
             machine,
             blob: Arc::clone(&self.blob),
-            since: read_cap.since,
             explicitly_expired: false,
         };
 
         Ok((writer, reader))
     }
 
+    /// Establishes and returns a new `SinceHandle` as the currently active
+    /// since handle for this shard. This will prevent `SinceHandles` with a
+    /// different id from calling [`SinceHandle::downgrade_since`], that is, we
+    /// are fencing them off.
+    // WIP: When would we return InvalidUsage here?
```

Comment:
do we return InvalidUsage? I can't initially see why we would

Reply:
We do, because creating the `Machine` can fail with `InvalidUsage` if the codec types don't match. We have codec types on `SinceHandle` because `Machine` needs them, even though it would be better for users if `SinceHandle` didn't have them.

If we want to change that, we'd have to remove the codec types from `Machine` and instead perform the checks in the read/write handles. Perhaps we should do that anyways, WDYT?

Reply:
we at least have to have a <T> on State and Machine for all the stuff with Antichains and Descriptions to work, which means we'd have to split up the checking for codec match into two places. could be fine, but I don't think it's an unambiguous win